test/unit/adversarial/helper.py

```diff
@@ -161,7 +162,7 @@ class initializers:  # noqa: D106
             class GlorotNormal:  # noqa: D106
                 pass
 
-        Model = type("Model", (model_class,), {})
+        Model = Model
```

Comment:
O_o

Reply:
oh this is definitely questionable. maybe it can be replace with a function so it is more clear for both assignments? it is necessary since Pickle needs to know about `Model` outside of all the nested classes, otherwise it doesn't recognise it. something to do with the nested classes from what I read. but I agree it is not a great way to do things.

Reply:
@adrinjalali I need some help in shaping this is a way that is a bit better. I have tried multiple things, but picke rejects everything except defining this value at a top level and using the same object later in the class to assign it to the class attribute. Anything I can do syntax-wise to make this a bit more acceptable? Or do you have some other ideas?

Reply:
Can you explain what the purpose of this here is? This is only for testing purposes, so it doesn't matter how dirty it is I'd say.

Reply:
Yes! I don't keep any references to the docs I read at the time when I fixed this (I cherrypicked it from an old branch) so I will try to summarise what I remember.
- The issue: the pickle related estimator checks complain that they can't pickle because they cannot find the `Model` attribute.
- Reading docs I found out that the deep nested classes are an antipattern for pickle, as in it doesn't deal with them well and there are often pickling issues when used. 
- Things I remember trying: rewriting a part of that nested class or reducing the nesting, moving the attribute position around (it had too big implications on the rest of the code and how we test and if I was to make it work I had to rewrite much more), I tried having it twice(top and nested level) assigning from a getter kind of function so it's "nicer" (complained it isn't the same object). 
- The only thing that worked was defining it like this top level and just using it as a reference in the assignment in the nested class. I am also not found of the solution, but I don't know pickle well enough to think of an alternative.